cmd/internal/backup/backup.go

```diff
@@ -75,3 +41,47 @@ func Start(ctx context.Context, log *zap.SugaredLogger, backupSchedule string, d
 	c.Stop()
 	return nil
 }
+
+func CreateBackup(log *zap.SugaredLogger, db database.DatabaseProber, bp backuproviders.BackupProvider, metrics *metrics.Metrics, comp *compress.Compressor) error {
+	err := db.Backup()
```

Comment:
This must be guarded with a lock to prevent concurrent backup execution which could lead to corrupt data in the backup

Reply:
Done with a semaphore.